File: src/main/java/com/donniemattingly/kotlinbrownbag/RecipeController.java
package com.donniemattingly.kotlinbrownbag;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
public class RecipeController {

  private final RecipeService recipeService;

  @Autowired
  public RecipeController(RecipeService recipeService) {
    this.recipeService = recipeService;
  }

  @GetMapping("/recipes")
  public List<Recipe> getRecipes() {
    return this.recipeService.getRecipes();
  }

  @GetMapping("/best_recipe")
  public Recipe getBestRecipes() {
    return this.recipeService.bestRecipe().orElse(null);
  }
}

File: src/main/java/com/donniemattingly/kotlinbrownbag/RecipeController.kt
package com.donniemattingly.kotlinbrownbag


import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class RecipeController @Autowired
constructor(private val recipeService: RecipeService) {

    val recipes: List<Recipe>
        @GetMapping("/recipes")
        get() = this.recipeService.recipes

    val bestRecipes: Recipe
        @GetMapping("/best_recipe")
        get() = this.recipeService.bestRecipe().orElse(null)
}
